tokens.js

```diff
@@ -0,0 +1,33 @@
+require('dotenv').config()
+
+// https://stackoverflow.com/a/69058154/1413302
+const isTokenExpired = (token) => (Date.now() >= JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString()).exp * 1000)
+
+/**
+ * Use the privately stored refresh token to generate a new access token for
+ * your RERUM-connected application. There is no way to authenticate this 
+ * process, so protect your refresh token and replace it if it is exposed. 
+ */
+function generateNewAccessToken() {
+    const payload = {
+        methdo: 'POST',
+        body: JSON.stringify({ refresh_token: process.env.refresh_token })
+    }
+    fetch(process.env.RERUM_ACCESS_TOKEN_URL, payload)
+        .then(response => {
+            if (response.ok) {
+                return response.json()
+            }
+            return Promise.reject(response)
+        })
+        .then(tokenObject => process.env.access_token = tokenObject.access_token)
+        .catch(err => console.error("Token not updated: ", err))
+    console.warn("Access Token expired. Consider updating your .env files")
+}
+
+/**
+ * This will conduct a simple check against the expiry date in your token.
+ * This does not validate your access token, so you may still be rejected by 
+ * your RERUM instance as unauthorized.
+ */
+if (isTokenExpired(process.env.access_token)) { generateNewAccessToken() }
```

Comment:
export would expose it to other modules. This code runs when it is imported/required, since it is not within a definition. Since all I use are environmental variables, it has no dependencies to run and should just go as soon as it can.

If we wind up following my advice and creating a check that can be called inside a 403 error, for example, that would require an `export`.